handbook/engineering/distribution/observability/monitoring_architecture.md

```diff
@@ -0,0 +1,114 @@
+
+**Note:** Looking for _how to develop Sourcegraph monitoring?_ See the [monitoring developer guide](monitoring.md).
+
+This document describes the architecture of Sourcegraph's monitoring stack, and the technical decisions we have made to date and why.
+
+<!-- generated from monitoring_architecture.excalidraw -->
+![architecture diagram](https://storage.googleapis.com/sourcegraph-assets/monitoring-architecture.png)
+
+## Long-term vision
+
+To better understand our goals with Sourcegraph's monitoring stack, please read [monitoring pillars: long-term vision](monitoring_pillars.md#long-term-vision).
+
+## Monitoring generator
+
+We use a custom [declarative Go generator syntax](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/monitoring) for:
+
+- Defining the services we monitor.
+- Describing _what those services do_ to site admins.
+- Laying out dashboards in a uniform, consistent, and simple way.
+- Generating the [Prometheus alerting rules](#alerting) and Grafana dashboards.
+- Generating documentation in the form of ["possible solutions"](https://docs.sourcegraph.com/admin/observability/alert_solutions) for site admins to follow when alerts are firing.
+
+This allows us to assert constraints and principles that we want to hold ourselves to, as described in [our monitoring pillars](monitoring_pillars.md).
+
+To learn more about adding monitoring using the generator, see ["How easy is it to add monitoring?"](monitoring.md#how-easy-is-it-to-add-monitoring)
+
+## Sourcegraph deployment
+
+### Sourcegraph Grafana
+
+We use [Grafana](https://grafana.com) for:
+
+- Displaying generated dashboards for our Prometheus metrics and alerts.
+- Providing an interface to query Prometheus metrics and Jaeger traces.
+
+The [`sourcegraph/grafana`](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/grafana) image handles shipping Grafana and Sourcegraph monitoring dashboards. It bundles:
+
+* Preconfigured [Grafana](https://grafana.com), which displays data from Prometheus and Jaeger
+* Dashboards generated by the [monitoring generator](#monitoring-generator).
+
+#### Admin reverse-proxy
+
+For convenience, Grafana is served on `/-/debug/grafana` on all Sourcegraph deployments via a reverse-proxy restricted to admins.
+
+Services served via reverse-proxy in this manner could be vulnerable to [cross-site request forgery](https://owasp.org/www-community/attacks/csrf), which is complicated to resolve ([#6075](https://github.com/sourcegraph/sourcegraph/issues/6075)). This means that at the moment, making changes to Grafana using the Grafana UI is not possible without setting up a port-forward, something [we want to avoid asking customers to do](monitoring_pillars.md#long-term-vision). In addition, provisioned dashboards generated by the [monitoring generator](#monitoring-generator) cannot be edited at all.
```

Comment:
> [we want to avoid asking customers to do](monitoring_pillars.md#long-term-vision)

I dont think that sections describes why we want to prevent that. Its clear that we want to ship good defaults and dashboard that are similar to the ones we use internally. While I think its a good idea to provide the dashboards as a static asset of our product, I dont think preventing customization is necessarily a requirement because dogfooding.

Reply:
It's not about preventing customization being a requirement, but more that we want to remove the barriers to doing so, and in the past ConfigMap-based/port-forward-requiring approaches has lead to limited adoption of alerting (from what I understand - again, this comment as reference: https://github.com/sourcegraph/sourcegraph/issues/11452#issuecomment-648628953

Reply:
Ok, I got this the other way around. This is just describing that Grafana does not allow provisioned dashboards (the ones you can auto-load by adding them to `provisioning/dashboards`) to be edited. Is that correct?

Reply:
Yep! I tried to improve the wording slighly in 5bf3928686842cc335f303d69f45610afca098b3